source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -246,26 +244,20 @@ bool TrySetupContext(string kubeConfig, string @namespace, string accountType)
 
                 if (isUsingAzureServicePrincipalAuth)
                 {
-                    if (!TrySetAz())
-                    {
-                        log.Error("Could not find az. Make sure az is on the PATH.");
-                        return false;
-                    }
-
-                    ConfigureAzAccount();
+                    var azureCli = new AzureCli(log, commandLineRunner, workingDirectory, environmentVars);
+                    var azureAuth = new AzureKubernetesServicesAuth(azureCli, kubectlCli, variables);
 
-                    SetupContextForAzureServicePrincipal(kubeConfig, @namespace);
+                    if (!azureAuth.TryConfigure(@namespace, kubeConfig))
+                        return false;
```

Comment:
Thanks for extracting this 👍 I'm thinking it would be nice if we could extract all such authentication methods into their own classes and give them a common interface (the already extracted Azure Auth and GCloud Auth do demonstrate a common pattern). Then we could probably remove this large if-else statement here. Something like:
```cs
IAuthenticationProvider auth = AuthenticationProviderFactory.Create(variables, accountType);
return auth.Configure(variables, accountType);
```

But there are too many changes to make to do this, and additional risks, which make this probably not worthwhile. I'm happy the way it currently is.

Reply:
We're on the same wavelength :) I was just telling @IsaacCalligeros95 on Tuesday about my plans to do exactly this.

I think this would be a really positive direction for this code, but agree that the scope of refactoring is at its limit for one PR. I can add it as a TODO comment so that the next person to touch it works toward it, though.